### PR TITLE
[General] Logging & launching improvements

### PR DESCRIPTION
**Pull request by CommandMC**
Main changes:
- Logs are now live (meaning new output is written to the file immediately instead of waiting for game exit)
- If an error happens, part of the log is still written (as much info as possible is provided, to help debug the issue)
- Errors themselves are now logged with a stacktrace (not too helpful in production since function & file names are minified, but might help in development)
- The "Error log"/"Game log" (stdout & stderr) split was removed, it's both in "Game log" again now
- The log now contains a timestamp

Other changes:
- `getRunnerCallWithoutCredentials` is exported again since it's needed for the new log behavior
- Some code cleanup, didn't bother to open a separate PR for just that
---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [X] Tested the feature and it's working on a current and clean install.
- [X] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: electron/games.ts [hunk 1]
```diff
@@ -20,26 +19,32 @@ abstract class Game {
       return GOGGame.get(appName)
     }
   }
+  public get logFileLocation() {
+    return join(
+      heroicGamesConfigPath,
+      `${this.appName.split(' ')[0]}-lastPlay.log`
```
Comment (Nocccer) [inline electron/games.ts:25]:
```suggestion
      `${this.appName}-lastPlay.log`
```

File: electron/gog/games.ts [hunk 2]
```diff
@@ -392,35 +395,43 @@ class GOGGame extends Game {
...
+      wrappers,
+      join(...Object.values(getGOGdlBin()))
     )
+    appendFileSync(
+      this.logFileLocation,
+      `Launch Command: ${fullCommand}
+
+Game Log:
+`
+    )
```
Comment (Nocccer) [inline electron/gog/games.ts:410]: Maybe this is better readable?
```suggestion
    appendFileSync(
      this.logFileLocation,
      [`LAUNCH COMMAND: ${fullCommand}`, 'GAME LOG:'].join('\n'))
```

File: electron/legendary/games.ts [hunk 3]
```diff
@@ -659,15 +657,29 @@ class LegendaryGame extends Game {
...
+      commandEnv,
+      wrappers,
+      join(...Object.values(getLegendaryBin()))
+    )
+    appendFileSync(
+      this.logFileLocation,
+      `Launch Command: ${fullCommand}
+
+Game Log:
+`
```
Comment (Nocccer) [inline electron/legendary/games.ts:671]: Same here. Maybe we think of a layout for games log aswell? Like general log has it? 
Calculated tabs/spaces etc.

File: electron/main.ts [hunk 4]
```diff
@@ -814,60 +813,50 @@ ipcMain.handle(
...
+    writeFileSync(
+      game.logFileLocation,
+      `System Info:
 ${await getSystemInfo()}
 
-Game Settings: ${JSON.stringify(gameSettings, null, '\t')}
+Game Settings: ${JSON.stringify(await game.getSettings(), null, '\t')}
+
+Game launched at: ${startPlayingDate}
+
```
Comment (Nocccer) [inline electron/main.ts:834]: Don't like the way of writing this multiline strings. There should be a better way like string arrray etc.

E.g.:

```typescript
['first line', 'second line'].join('\n')
```

**Review by Nocccer — CHANGES_REQUESTED** [review]
Have some suggestions and ideas we need to talk about.

File: electron/games.ts (hunk 1 above)
Comment (CommandMC) [reply electron/games.ts:25]: Well, this was how the filename was originally generated. But I do agree that the split is useless, AppNames can't have spaces

File: electron/legendary/games.ts (hunk 3 above)
Comment (CommandMC) [reply electron/legendary/games.ts]: In terms of a "layout" for these logs: I'd say the current format is *fine* for what it needs to do. No need to over-complicate it

File: electron/legendary/games.ts [hunk 5]
```diff
@@ -665,10 +665,7 @@ class LegendaryGame extends Game {
     )
     appendFileSync(
       this.logFileLocation,
-      `Launch Command: ${fullCommand}
-
-Game Log:
-`
+      `Launch Command: ${fullCommand}\n\nGame Log:\n`
```
Comment (Nocccer) [inline electron/legendary/games.ts:668]:
```suggestion
    `Launch Command: ${fullCommand}\n` + 
          '\n' + 
          'Game Log:\n'
```

**Review by Nocccer — CHANGES_REQUESTED** [review]
I left some suggestions. For me it is important to have readable strings in the source code. A string with 1-2 linebreaks is maybe readable and therefore fine, but multine strings should be fast readable, especially with alot of linebreaks.

File: electron/legendary/games.ts (hunk 5 above)
Comment (CommandMC) [reply electron/legendary/games.ts:668]: Linting automatically changes this to `` `Launch Command: ${fullCommand}\n` + '\n' + 'Game Log:\n' ``

**Comment by CommandMC** [comment]
> What do you think?

I honestly don't really mind. Whatever works works for me, if you want a specific style then I'll use that specific style

**Review by Nocccer — APPROVED** [review]
Sorry for this back adn forth.
I think the best way is to write code always in mind someone else needs to understand it possible with the least effort. So readable code is the key.

But LGTM now.

**Pull request merged**